src/core/qgsconnectionpool.h

```diff
@@ -76,13 +68,28 @@ class QgsConnectionPoolGroup
     {
     }
 
-    ~QgsConnectionPoolGroup()
-    {
-      for ( const Item &item : std::as_const( conns ) )
-      {
-        qgsConnectionPool_ConnectionDestroy( item.c );
-      }
-    }
+    virtual ~QgsConnectionPoolGroup() = 0;
+
+    /**
+     * Classes that implement this template must implement their own connection functions.
+     * Create a connection \a c based on connection information \a connInfo.
+     */
+    virtual void connectionCreate( const QString &connInfo, T &c ) = 0;
```

Comment:
... and other occurrences too
```suggestion
    virtual void connectionCreate( const QString &connInfo, T &connection ) = 0;
```